src/vmm/src/device_manager/persist.rs

```diff
@@ -203,14 +203,11 @@ impl<'a> Persist<'a> for MMIODeviceManager {
                     });
                 }
                 TYPE_BLOCK => {
-                    let block_state = locked_device
-                        .as_any()
-                        .downcast_ref::<Block>()
-                        .unwrap()
-                        .save();
+                    let block = locked_device.as_mut_any().downcast_mut::<Block>().unwrap();
+                    block.prepare_save();
```

Comment:
we could move this logic to the `save()` impl of the Block device, so that we get better encapsulation (and make prepare_save() private).
This way, users of the block device would not forget to call `prepare_save` before `save`

Reply:
The problem is that save takes an immutable reference to self and we need a mutable one. That's why I called `prepare_save` here. I don't think it's worth trying to modify the signature of `save`.

Reply:
oh I see. In this case it's not worth it indeed